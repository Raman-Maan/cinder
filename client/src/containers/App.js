import React, { Component } from 'react';
import logo from '../images/logo.svg';
import './App.css';

class App extends Component {
  state = {
    server_response: ''
  };

  getMessage = () => {
    return fetch('/api/user/example')
      .then(res => res.json())
      .catch(err => console.error(err));
  };

  componentDidMount() {
    this.getMessage()
      .then(msg => this.setState({ server_response: msg.example }))
      .catch(err => console.error(err));
  }

  render() {
    return (
      <div className="App">
        <header className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <h1 className="App-title">Welcome to React</h1>
        </header>
        <p className="App-intro">{this.state.server_response}</p>
      </div>
    );
  }
}

export default App;
