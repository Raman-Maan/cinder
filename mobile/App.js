import React from 'react';
import Expo, { AppLoading } from 'expo';

import Home from './containers/home';

export default class App extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      fontLoaded: false,
    };
  }

  componentDidMount() {
    this.loadAssets();
  }

  async loadAssets() {
    await Expo.Font.loadAsync({
      Roboto: require('native-base/Fonts/Roboto.ttf'),
      Roboto_medium: require('native-base/Fonts/Roboto_medium.ttf'),
    });

    this.setState({ fontLoaded: true });
  }

  render() {
    if (!this.state.fontLoaded) {
      return <AppLoading />;
    }

    return (
      <Home />
    );
  }
}
