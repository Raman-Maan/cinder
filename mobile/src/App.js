import React from 'react';
import { View, StatusBar } from 'react-native';
import { StackNavigator } from 'react-navigation';

import Home from './containers/home';
import Profile from './containers/blank';
import Matches from './containers/matches';

const RootStack = StackNavigator({
  Home: {
    screen: Home,
  },
  Profile: {
    screen: Profile,
  },
  Matches: {
    screen: Matches,
  },
}, {
  initialRouteName: 'Matches',
  navigationOptions: {
    title: 'cinder',
  },
});

export default class App extends React.Component {
  render() {
    return (
      <RootStack />
    );
  }
}
