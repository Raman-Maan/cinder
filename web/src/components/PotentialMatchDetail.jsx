import React from 'react';
import { Button } from 'reactstrap';
import './styles/PotentialMatchDetail.css';

export default class PotentialMatchDetail extends React.Component {
  render() {
    if (!this.props.potentialMatchDetail) {
      return (
        <div className="UserDetailEmpty">
          <div className="empty">
            <h3 className="msg align-self-center">
              There's no one new around you.
            </h3>
          </div>
        </div>
      );
    }

    return (
      <div className="UserDetail">
        <img
          className="userImage"
          src={this.props.potentialMatchDetail.img}
          alt=""
        />
        <br />
        <div className="UserInfo">
          <span className="UserNameAndAge">
            {this.props.potentialMatchDetail.title},
            {this.props.potentialMatchDetail.age}
          </span>
          <br />
          <span className="UserBio">
            {this.props.potentialMatchDetail.subtitle}
          </span>
        </div>
        <div className="buttons">
          <Button
            className="NotButton"
            color="danger"
            size="lg"
            onClick={this.props.handlePass}
          >
            NOT
          </Button>{' '}
          <Button
            className="HotButton"
            color="success"
            size="lg"
            onClick={this.props.handleLike}
          >
            HOT
          </Button>
        </div>
      </div>
    );
  }
}
