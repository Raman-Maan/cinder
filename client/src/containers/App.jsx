import React, { Component } from 'react';
import logo from '../assets/logo.svg';
import './App.css';

class App extends Component {
  static getMessage() {
    return fetch('/api/user/example')
      .then(res => res.json())
      .catch(err => console.error(err));
  }

  constructor() {
    super();
    this.state = {
      server_response: 'No request for message has been made'
    };
  }

  componentDidMount() {
    App.getMessage()
      .then(msg => this.setState({ server_response: msg.example }))
      .catch(err => console.error(err));
  }

  render() {
    return (
      <div className="App">
        <header className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <h1 className="App-title">Welcome to React</h1>
        </header>
        <p className="App-intro">{this.state.server_response}</p>
      </div>
    );
  }
}

export default App;
