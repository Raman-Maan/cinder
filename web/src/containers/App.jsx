import React, { Component } from 'react';
import { Switch, Route, withRouter } from 'react-router-dom';
import PrivateRoute from '../components/PrivateRoute';
import './styles/App.css';
import { connect } from 'react-redux';

import Layout from './Layout';
import Login from './Login';
import Signup from './Signup';

class App extends Component {
  render() {
    return (
      <div className="App">
        <Switch>
          <Route exact path="/login" component={Login} />
          <PrivateRoute
            exact
            path="/"
            authed={this.props.isAuthenticated}
            component={Layout}
          />
          <Route exact path="/signup" component={Signup} />
        </Switch>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  isAuthenticated: state.auth.isAuthenticated
});

export default withRouter(connect(mapStateToProps, null)(App));
