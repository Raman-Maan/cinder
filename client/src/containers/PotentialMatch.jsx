import React, { Component } from 'react';
import PotentialMatchDetail from '../components/PotentialMatchDetail';

export default class PotentialMatch extends Component {
  constructor(props) {
    super(props);
    this.fetchPotentialMatches = this.fetchPotentialMatches.bind(this);
    this.handleApprovedMatch = this.handleApprovedMatch.bind(this);
    this.handleRejectedMatch = this.handleRejectedMatch.bind(this);

    this.state = {
      potentialMatches: [],
      matchIndex: 0
    };
  }

  componentDidMount() {
    this.fetchPotentialMatches();
  }

  increment() {
    this.setState(prevIndex => {
      return { matchIndex: prevIndex.matchIndex + 1 };
    });
  }

  handleApprovedMatch() {
    this.increment();
    // api call to add to like table, sending the user id of the approved user
  }

  handleRejectedMatch() {
    this.increment();
    // api call to add to like table, sending the user id of the rejected user
  }

  fetchPotentialMatches() {
    this.setState({
      potentialMatches: [
        {
          id: 0,
          title: 'Kendrick Lamar',
          age: 30,
          subtitle: 'How u doin?',
          img: 'http://cache.umusic.com/_sites/kendricklamar.com/images/og.jpg'
        },
        {
          id: 1,
          title: 'Mac Miller | Larry Fisherman | Delusional Thomas',
          age: 26,
          subtitle:
            ';););) Want to meet hot new singles? Click this totally legit link: www.gethotsingles.com',
          img:
            'https://i.scdn.co/image/f4509fe9c589c12be5470653178f901bd697b97b'
        },
        {
          id: 2,
          title: 'Ian Simpson',
          age: 21,
          subtitle: 'What u wearin',
          img:
            'https://media.pitchfork.com/photos/592997c25e6ef9596931f65a/1:1/w_300/e2fc485c.jpg'
        }
      ]
    });
  }

  render() {
    if (!this.state.potentialMatches.length) {
      return (
        <div>
          <h1>There's no one new around you.</h1>
        </div>
      );
    }

    return (
      <div className="PotentialMatchDetailContainer">
        <PotentialMatchDetail
          potentialMatchDetail={
            this.state.potentialMatches[this.state.matchIndex]
          }
          handlePass={this.handleRejectedMatch.bind(this)}
          handleLike={this.handleApprovedMatch.bind(this)}
        />
        <br />
      </div>
    );
  }
}
