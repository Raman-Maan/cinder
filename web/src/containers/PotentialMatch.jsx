import React, { Component } from 'react';
import PotentialMatchDetail from '../components/PotentialMatchDetail';
import Auth from '../utils/authService';

export default class PotentialMatch extends Component {
  constructor(props) {
    super(props);
    this.fetchPotentialMatches = this.fetchPotentialMatches.bind(this);
    this.handleLike = this.handleLike.bind(this);
    this.handlePass = this.handlePass.bind(this);

    this.state = {
      potentialMatches: [],
      matchIndex: 0
    };
  }

  componentDidMount() {
    this.fetchPotentialMatches();
  }

  incrementPotentialMatchIndex() {
    this.setState(prevIndex => ({ matchIndex: prevIndex.matchIndex + 1 }));
  }

  handlePass() {
    this.submitUserAction('pass');
    this.incrementPotentialMatchIndex();
  }

  handleLike() {
    this.submitUserAction('like');
    this.incrementPotentialMatchIndex();
  }

  submitUserAction(userAction) {
    const matchedUser = this.state.potentialMatches[this.state.matchIndex].id;

    fetch(`/api/users/${Auth.loggedInUser.id}/matches/${matchedUser}/${userAction}`, { method: 'POST' })
      .then(res => res.json())
      .then(res => {
        this.setState({
          // TODO
        });
      })
      .catch(err => console.error(err));
  }

  fetchPotentialMatches() {
    fetch(`/api/users/${Auth.loggedInUser.id}/potentials`)
      .then(res => res.json())
      .then(res => {
        this.setState({
          potentialMatches: res.map(x => ({
            id: x.userID,
            title: x.userName,
            age: x.age,
            img: x.primaryPic
          }))
        });
      })
      .catch(err => console.error(err));
  }

  render() {
    if (!this.state.potentialMatches.length) {
      return (
        <div>
          <h1>There's no one new around you.</h1>
        </div>
      );
    }

    return (
      <div className="PotentialMatchDetailContainer">
        <PotentialMatchDetail
          potentialMatchDetail={
            this.state.potentialMatches[this.state.matchIndex]
          }
          handlePass={this.handlePass.bind(this)}
          handleLike={this.handleLike.bind(this)}
        />
        <br />
      </div>
    );
  }
}
