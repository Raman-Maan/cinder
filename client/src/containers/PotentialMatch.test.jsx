import React from 'react';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import sinon from 'sinon';

import App from './App';
import PotentialMatch from './PotentialMatch';
import PotentialMatchDetail from '../components/PotentialMatchDetail';

describe('<PotentialMatch />', () => {
  const testPotentialMatches = [
    {
      userName: 'Some Name 1',
      userID: 1,
      age: 21,
      matchDate: '2018-02-03T18:09:05.000Z',
      primaryPic: null
    },
    {
      userName: 'Some Name 2',
      userID: 5,
      age: 22,
      matchDate: '2018-01-03T18:09:05.000Z',
      primaryPic: 'http://c.min.ms/m/b/13/13022/e8250eba.jpeg'
    },
    {
      userName: 'Some Name 3',
      userID: 3,
      age: 23,
      matchDate: '2018-02-06T18:09:05.000Z',
      primaryPic: null
    },
    {
      userName: 'Some Name 4',
      userID: 2,
      age: 24,
      matchDate: '2018-02-04T18:09:05.000Z',
      primaryPic: 'http://c.min.ms/m/b/13/13022/e8250eba.jpeg'
    }
  ];

  //TODO - implement network test when hooked up to end point for fetching potential matches
  // beforeEach(() => {
  //     fetchMock.reset();
  //     fetchMock.restore();
  //     fetchMock.get(/\/api\/users\/[0-9]*\/matches/, testPotentialMatches);
  // });

  // afterAll(() => {
  //     fetchMock.reset();
  //     fetchMock.restore();
  // });

  it('renders without crashing', () => {
    // simple smoke test
    shallow(<PotentialMatch />);
    mount(<PotentialMatch />);
  });

  it('renders correctly', () => {
    const component = renderer
      .create(<PotentialMatch potentialMatches={testPotentialMatches} />)
      .toJSON();

    expect(component).toMatchSnapshot();
  });

  it('renders with empty props correctly', () => {
    const component = renderer.create(<PotentialMatch />).toJSON();
    expect(component).toMatchSnapshot();
  });
});
