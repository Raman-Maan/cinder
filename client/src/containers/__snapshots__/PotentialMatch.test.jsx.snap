// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PotentialMatch /> renders with empty props correctly 1`] = `
<div>
  <h1>
    There's no one new around you.
  </h1>
</div>
`;
